### DISCUSSION TITLE
Declare indexes while keeping Pydantic and SQLModel logic separated

### DISCUSSION THREAD

**Discussion by choucavalier** [fastapi/sqlmodel#1001] [Questions] [answered]
### Example Code

```python
from typing import Optional

from sqlmodel import Field as SQLField
from sqlmodel import SQLModel
from pydantic import BaseModel

class MyType(BaseModel):
    some_attr: Optional[str] = None
    some_other_attr: int = None


class MyTypeSQL(MyTypeBase, SQLModel, table=True):
    id: Optional[int] = SQLField(default=None, primary_key=True)
```

### Description

How can I declare `some_other_attr` as a field that should be used for indexing, while still keeping the pydantic and SQLModel logic separated?

**Comment by ixycoexzckwpmlcu** [comment, marked as answer]
Just curious - why would you like to separate your Pydantic logic and SQLModel logic? Non-table SQLModel classes (i.e., without `table=true`) won't create tables and will effectively function as Pydantic models (more in the [FastAPI and Pydantic Tutorial](https://sqlmodel.tiangolo.com/tutorial/fastapi/), in particular the [Multiple Models tutorial](https://sqlmodel.tiangolo.com/tutorial/fastapi/multiple-models/)).

Adapted to your example, this would look like the following:
```py
from typing import Optional

from sqlmodel import Field, SQLModel

class MyType(SQLModel):
    some_attr: Optional[str] = Field(index=True)
    some_other_attr: int = None
    

class MyTypeSQL(MyType, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
```
If using SQLite (for example), this code would create the following table:
```sql
CREATE TABLE mytypesql (
	some_attr VARCHAR, 
	some_other_attr INTEGER NOT NULL, 
	id INTEGER NOT NULL, 
	PRIMARY KEY (id)
);
CREATE INDEX ix_mytypesql_some_attr ON mytypesql (some_attr);
```
